OCRSpace/operations.py

```diff
@@ -0,0 +1,137 @@
+""" Copyright start
+  Copyright (C) 2008 - 2021 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import requests
+import subprocess
+
+from connectors.cyops_utilities.builtins import download_file_from_cyops
+from integrations.crudhub import make_request
+from connectors.core.connector import get_logger, ConnectorError
+from os.path import join
+
+logger = get_logger('ocr-space')
+
+
+class OCRSpace(object):
+    def __init__(self, config):
+        self.log = logger
+        self.maximum_file_size = 32000000
+        self.base_url = config.get('server').strip() + '/Parse/Image'
+        if not self.base_url.startswith('https://'):
+            self.base_url = 'https://' + self.base_url
+        self.api_key = config.get('api_key')
+        self.verify_ssl = config.get('verify_ssl')
+        self.payload = {'isOverlayRequired': 'True', 'apikey': self.api_key, 'language': 'eng'}
+        self.__setupSession()
+
+    def __setupSession(self):
+        self.log.info('Setup session')
+        self.session = requests.Session()
+        self.session.verify = self.verify_ssl
+        self.session.headers.update({'accept': 'application/json'})
+
+    def __postUrl(self, params={}, data={}, files=None):
+        try:
+            self.log.info('Params: {}'.format(params))
+            if files is not None:
+                self.log.info('Posting files.')
+            if data:
+                self.log.info('Posting data.')
+                self.payload.update(data)
+            url = self.base_url
+            self.log.info('Posting to URL: {}'.format(url))
+            res = self.session.post(url, params=params, data=self.payload, files=files)
+            if res.status_code == 204:
+                return {'message': 'Request rate limit exceeded. You are making more requests than allowed.'}
+            return res.json()
+        except Exception as Err:
+            raise ConnectorError(Err)
+
+    def submitFile(self, file_iri, data):
+        try:
+            file_path = join('/tmp', download_file_from_cyops(file_iri)['cyops_file_path'])
+            logger.info(file_path)
+            out = subprocess.Popen(["file", str(file_path)], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Comment:
for this part please refer [(https://github.com/fortinet-fortisoar/connector-hybrid-analysis/blob/develop/hybrid-analysis/operations.py)]

Reply:
Thanks for providing this reference. I went through it and observe that most of the solution in this code is similar to the given reference except subprocess command I have used in line #57. My requirement is to automatically detect the type of file given by the user and I don't find anything related to this in the referred module.
Can you please specifically point out what you want me to change here?